core/services/keeper/registry_interface.go

```diff
@@ -119,8 +126,36 @@ func (rw *RegistryWrapper) GetActiveUpkeepIDs(opts *bind.CallOpts) ([]*big.Int,
 		}
 		return activeUpkeeps, nil
 	case RegistryVersion_1_2:
-		// TODO (sc-37024): Get active upkeep IDs from contract in batches
-		return rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(0), big.NewInt(0))
+		// fetch the current block number so batched GetActiveUpkeepIDs calls can be performed on the same block
+		header, err := rw.evmClient.HeaderByNumber(context.Background(), nil)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to fetch EVM block header")
+		}
+		if opts != nil {
+			opts.BlockNumber = header.Number
+		} else {
+			opts = &bind.CallOpts{
+				BlockNumber: header.Number,
+			}
+		}
+
+		state, err := rw.contract1_2.GetState(opts)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get contract state at block number %d", header.Number.Int64())
+		}
+
+		activeUpkeepIDs := make([]*big.Int, 0)
+		for int64(len(activeUpkeepIDs)) < state.State.NumUpkeeps.Int64() {
+			startIndex := int64(len(activeUpkeepIDs))
+			maxCount := int64(math.Min(float64(ActiveUpkeepIDBatchSize), float64(state.State.NumUpkeeps.Int64()-int64(len(activeUpkeepIDs)))))
```

Comment:
why use float64 here instead of int64?

Reply:
ah i see it's for math.min. Not sure if there are consequences of converting to float6, i would just prefer an extra conditional statement while keeping everything int64

Reply:
yes this is for math.Min. it's frustrating there is no generics yet on this func.
changed to use if statement.